src/shipit_pulse_listener/shipit_pulse_listener/listener.py

```diff
@@ -141,7 +145,23 @@ def __init__(self, configuration):
     def is_coverage_task(self, task):
         return task['task']['metadata']['name'].startswith('build-linux64-ccov')
 
+    def as_utc(self, d):
+        if d.tzinfo is None or d.tzinfo.utcoffset(d) is None:
+            return pytz.utc.localize(d)
+        return d.astimezone(pytz.utc)
+
+    def is_old_task(self, task):
+        for run in task['status']['runs']:
+            run_date = self.as_utc(dateutil.parser.parse(run['resolved']))
+            if run_date < self.as_utc(datetime.utcnow() - timedelta(1)):
```

Comment:
Why compare dates in UTC format when you could compare integer timestamps?

Reply:
`run['resolved']` is a string like `2017-08-03T19:32:04.855Z`, so instead of turning it into a timestamp, I'm parsing it as a datetime object and then comparing it with `datetime.utcnow() - timedelta(1)`. I need to set the date as UTC because the datetime object `datetime.utcnow() - timedelta(1)` doesn't have timezone information (and so can't be compared with `run_date`).

I'm doing it this way just for personal preference, as I find it easily readable.